File: labs/6/BookReviews/app/src/main/java/me/henryfbp/sqlitelab/Book.java
package me.henryfbp.sqlitelab;

public class Book {
    private Long id;

    private String title;

    private String author;

    public Book() {
    }

    public Book(String title, String author) {
        super();
        this.title = title;
        this.author = author;
    }

    public Long getId() {
        return id;
    }

    public Book setId(Long id) {
        this.id = id;
        return this;
    }

    public String getTitle() {
        return title;
    }

    public Book setTitle(String title) {
        this.title = title;
        return this;

    }

    public String getAuthor() {
        return author;
    }

    public Book setAuthor(String author) {
        this.author = author;
        return this;
    }
    //getters & setters

    @Override
    public String toString() {
        return String.format("%s [id=%02d, title=%s, author=%s]",
                getClass().getSimpleName(), id, title, author);
    }

    public Boolean equals(Book b) {
        return b.getTitle().equals(getTitle()) &&
                b.getAuthor().equals(getAuthor());
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof Book) {
            return ((Book) o).equals(this);
        } else {
            return o.equals(this);
        }
    }
}

File: labs/6/BookReviews/app/src/main/java/me/henryfbp/sqlitelab/Book.kt
package me.henryfbp.sqlitelab

class Book() {

    constructor(id: Long, title: String, author: String) : this() {
        this.id = id
        this.title = title
        this.author = author
    }

    constructor(title: String, author: String) : this() {
        this.title = title
        this.author = author
    }

    var id: Long? = null

    lateinit var title: String

    lateinit var author: String


    override fun toString(): String {
        return String.format("%s [id=%02d, title=%s, author=%s]",
                javaClass.simpleName, id, title, author)
    }

    fun equals(b: Book): Boolean? {
        return b.title.equals(title) &&
                b.author.equals(author) &&
                b.id?.equals(id) ?: true
    }

    override fun equals(other: Any?): Boolean {
        return when (other) {
            is Book -> {
                other.equals(this)!!
            }
            else -> {
                other == this
            }
        }
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + title.hashCode()
        result = 31 * result + author.hashCode()
        return result
    }
}
